### PR TITLE
Insert MSS clamping for traffic arriving services

### PR DESCRIPTION
**Pull request by mangelajo**
OVN load balancing does not propagate fragment ICMPs down to the
 services, so when fragmentation becomes necessary the TCP stack
 of the service pod won't be able to detect this situation.
    
 Until this is fixed in OVN this workaround (to be enhanced later)
 will fix TCP.
    
 Fixes-Issue: #1022
    
 Signed-off-by: Miguel Angel Ajo <majopela@redhat.com>

### CODE REVIEW

**Review by skitt — APPROVED** [review]

File: pkg/routeagent_driver/handlers/ovn/gateway_dataplane.go [hunk 1]
```diff
@@ -83,16 +84,31 @@ func (ovn *Handler) updateGatewayDataplane() error {
...
 
-	return [][]string{
+	rules := [][]string{
 		{"-i", ovnK8sGatewayInterface, "-o", ovn.cableRoutingInterface.Name, "-j", "ACCEPT"},
-		{"-i", ovn.cableRoutingInterface.Name, "-o", ovnK8sGatewayInterface, "-j", "ACCEPT"},
-	}, nil
+		{"-i", ovn.cableRoutingInterface.Name, "-o", ovnK8sGatewayInterface, "-j", "ACCEPT"}}
+
+	// NOTE: This is a workaround for submariner issue https://github.com/submariner-io/submariner/issues/1022
+	//		  1398 is 1500
```
Comment (sridhargaddam) [inline pkg/routeagent_driver/handlers/ovn/gateway_dataplane.go:105]: "1398 is 1500" -> Did you miss conveying something here?

**Comment by mangelajo** [comment]
one note after questions in slack:

   My plan is to implement PMTU discovery in 0.8.1 so we can use it here. Also... we need to go down to the OVN bug itself and get it fixed.

File: pkg/routeagent_driver/handlers/ovn/gateway_dataplane.go (hunk 1 above)
Comment (mangelajo) [reply pkg/routeagent_driver/handlers/ovn/gateway_dataplane.go:105]: oops

Comment (mangelajo) [reply pkg/routeagent_driver/handlers/ovn/gateway_dataplane.go]: Done.

**Review by tpantelis — APPROVED** [review]

**Review by sridhargaddam — APPROVED** [review]

**Pull request merged**